In [12]:
import numpy as np
import pylab as plt

%matplotlib inline

In [7]:
# collecting the probabilities

oldbagdict = {'brown':.3, 'yellow': .2, 'red':.2, 'green':.1, 'orange':.1, 'tan':.1, 'blue':0.}
newbagdict = {'brown':.13, 'yellow': .14, 'red':.13, 'green':.2, 'orange':.16, 'tan':0., 'blue':.24}

In [18]:
print('~~~Old Bag Chances~~~'+'\n')
print('H'+'\t'+'P(H)'+'\n')
for key,value in oldbagdict.items():
    print(str(key)+'\t'+str(value)+'\n')

~~~Old Bag Chances~~~

H	P(H)

orange	0.1

tan	0.1

red	0.2

blue	0.0

brown	0.3

green	0.1

yellow	0.2



In [19]:
print('~~~New Bag Chances~~~'+'\n')
print('H'+'\t'+'P(H)'+'\n')
for key, value in newbagdict.items():
    print(str(key)+'\t'+str(value)+'\n')

~~~New Bag Chances~~~

H	P(H)

orange	0.16

tan	0.0

red	0.13

blue	0.24

brown	0.13

green	0.2

yellow	0.14



Look what we found in this undated bag! A Yellow Piece!


In [21]:
print('Finding the Yellow Piece'+'\n')
print('H'+'\t'+'P(H)'+'\t'+'P(D|H)'+'\t'+'P(H)*P(D|H)')
print('New Bag'+'\t'+'.5'+'\t'+str(newbagdict['yellow'])+'\t'+str(.5*newbagdict['yellow']))
print('Old Bag'+'\t'+'.5'+'\t'+str(oldbagdict['yellow'])+'\t'+str(.5*oldbagdict['yellow']))

Finding the Yellow Piece

H	P(H)	P(D|H)	P(H)*P(D|H)
New Bag	.5	0.14	0.07
Old Bag	.5	0.2	0.1


In [22]:
print('(P(Yellow|Old Bag))/(P(Yellow|All)) = '+str(oldbagdict['yellow']/(newbagdict['yellow']+oldbagdict['yellow'])))

(P(Yellow|Old Bag))/(P(Yellow|All)) = 0.5882352941176471


In [23]:
print('Finding the Green Piece'+'\n')
print('H'+'\t'+'P(H)'+'\t'+'P(D|H)'+'\t'+'P(H)*P(D|H)')
print('New Bag'+'\t'+'.5'+'\t'+str(newbagdict['green'])+'\t'+str(.5*newbagdict['green']))
print('Old Bag'+'\t'+'.5'+'\t'+str(oldbagdict['green'])+'\t'+str(.5*oldbagdict['green']))

Finding the Green Piece

H	P(H)	P(D|H)	P(H)*P(D|H)
New Bag	.5	0.2	0.1
Old Bag	.5	0.1	0.05


In [24]:
print('(P(Green|New Bag))/(P(Green|All)) = '+str(oldbagdict['green']/(newbagdict['green']+oldbagdict['green'])))

(P(Green|New Bag))/(P(Green|All)) = 0.3333333333333333


In [25]:
print('(P(Yellow|Old Bag))*(P(Green|New Bag)) = '+str(newbagdict['green']*oldbagdict['yellow']))

(P(Yellow|Old Bag))*(P(Green|New Bag)) = 0.04000000000000001


In [26]:
print('(P(Green|Old Bag))*(P(Yellow|New Bag)) = '+str(oldbagdict['green']*newbagdict['yellow']))

(P(Green|Old Bag))*(P(Yellow|New Bag)) = 0.014000000000000002


Above are the relative probabilities!

In [27]:
print('(P(Yellow|Old Bag))*(P(Green|New Bag))/(P(Yellow)*P(Green)) = '+str(oldbagdict['green']*newbagdict['yellow']/(oldbagdict['green']*newbagdict['yellow']+oldbagdict['yellow']*newbagdict['green'])))

(P(Yellow|Old Bag))*(P(Green|New Bag))/(P(Yellow)*P(Green)) = 0.25925925925925924


In [28]:
print('Finding the Yellow Piece'+'\n')
print('H'+'\t'+'P(H)'+'\t'+'P(D|H)'+'\t'+'P(H)*P(D|H)'+'\t'+'P(H|D)')
print('New Bag'+'\t'+'.5'+'\t'+str(newbagdict['yellow'])+'\t'+str(.5*newbagdict['yellow'])+'\t'+str(newbagdict['yellow']/(newbagdict['yellow']+oldbagdict['yellow'])))
print('Old Bag'+'\t'+'.5'+'\t'+str(oldbagdict['yellow'])+'\t'+str(.5*oldbagdict['yellow'])+'\t'+str(oldbagdict['yellow']/(newbagdict['yellow']+oldbagdict['yellow'])))

Finding the Yellow Piece

H	P(H)	P(D|H)	P(H)*P(D|H)	P(H|D)
New Bag	.5	0.14	0.07	0.411764705882353
Old Bag	.5	0.2	0.1	0.5882352941176471


In [29]:
print('Finding the Green Piece'+'\n')
print('H'+'\t'+'P(H)'+'\t'+'P(D|H)'+'\t'+'P(H)*P(D|H)'+'\t'+'P(H|D)')
print('New Bag'+'\t'+'.5'+'\t'+str(newbagdict['green'])+'\t'+str(.5*newbagdict['green'])+'\t'+str(newbagdict['green']/(newbagdict['green']+oldbagdict['green'])))
print('Old Bag'+'\t'+'.5'+'\t'+str(oldbagdict['green'])+'\t'+str(.5*oldbagdict['green'])+'\t'+str(oldbagdict['green']/(newbagdict['green']+oldbagdict['green'])))

Finding the Green Piece

H	P(H)	P(D|H)	P(H)*P(D|H)	P(H|D)
New Bag	.5	0.2	0.1	0.6666666666666666
Old Bag	.5	0.1	0.05	0.3333333333333333


In [35]:
newgivegreen = newbagdict['green']/(newbagdict['green']+oldbagdict['green'])
oldgivegreen = oldbagdict['green']/(newbagdict['green']+oldbagdict['green'])
newgiveyellow = newbagdict['yellow']/(newbagdict['yellow']+oldbagdict['yellow'])
oldgiveyellow = oldbagdict['yellow']/(newbagdict['yellow']+oldbagdict['yellow'])
print('Finding Green and Yellow Pieces in Those Bags! Where Did Ya Find Those Round Beans?!')

print('Case'+'\t'+'\t'+'Not Normalized'+'\t'+'Normalized')
print('P(G|N)*P(Y|O)'+'\t'+str(newgivegreen*oldgiveyellow)+'\t'+str(newgivegreen*oldgiveyellow/(newgivegreen*oldgiveyellow+oldgivegreen*newgiveyellow)))
print('P(G|N)*P(Y|O)'+'\t'+str(oldgivegreen*newgiveyellow)+'\t'+str(oldgivegreen*newgiveyellow/(newgivegreen*oldgiveyellow+oldgivegreen*newgiveyellow)))


Finding Green and Yellow Pieces in Those Bags! Where Did Ya Find Those Round Beans?!
Case		Not Normalized	Normalized
P(G|N)*P(Y|O)	0.39215686274509803	0.7407407407407407
P(G|N)*P(Y|O)	0.13725490196078433	0.2592592592592593


The probability that they yellow M&M came from the yellow bag is .741

In [34]:
newgivegreen, newgiveyellow, oldgivegreen, oldgiveyellow

(0.6666666666666666, 0.411764705882353, 0.3333333333333333, 0.5882352941176471)